### PR TITLE
send the link to the email for password reset by default

### PR DESCRIPTION
**Pull request by lbalmaceda**

### CODE REVIEW

File: lock/src/main/java/com/auth0/lock/Lock.java [hunk 1]
```diff
@@ -134,6 +135,7 @@
         this.fullScreen = false;
         this.signUpEnabled = true;
         this.changePasswordEnabled = true;
+        this.requirePasswordOnPasswordReset = false;
```
Comment (hzalaz) [inline lock/src/main/java/com/auth0/lock/Lock.java]: `legacyPasswordReset`

File: lock/src/main/java/com/auth0/lock/Lock.java [hunk 2]
```diff
@@ -304,6 +310,16 @@ public boolean isChangePasswordEnabled() {
     }
 
     /**
+     * If Lock will ask for the new password when resetting the old one (previous behaviour)
```
Comment (hzalaz) [inline lock/src/main/java/com/auth0/lock/Lock.java]: Mark it as deprecated

File: lock/src/main/java/com/auth0/lock/fragment/DatabaseChangePasswordFragment.java [hunk 3]
```diff
@@ -51,6 +50,7 @@
 public class DatabaseChangePasswordFragment extends BaseTitledFragment {
 
     private static final String TAG = DatabaseChangePasswordFragment.class.getName();
+    public static final String REQUIRE_PASSWORD_NOW = "REQUIRE_PASSWORD_NOW";
```
Comment (hzalaz) [inline lock/src/main/java/com/auth0/lock/fragment/DatabaseChangePasswordFragment.java]: `USE_LEGACY_RESET_PASSWORD`

**Comment by hzalaz** [comment]
Please add a comment with a screenshot on how it looks

**Comment by lbalmaceda** [comment]
Nexus 5x:
nexus 5x

Nexus 6P:
nexus 6p

**Pull request merged**